lib/addressable/uri.rb

```diff
@@ -1132,7 +1141,9 @@ def normalized_host
         end
       end
       # All normalized values should be UTF-8
-      @normalized_host.force_encoding(Encoding::UTF_8) if @normalized_host
+      unless @normalized_host.empty?
+        @normalized_host.force_encoding(Encoding::UTF_8)
+      end
```

Comment:
Is it guaranteed that `@normalized_host` will always respond to `#empty?`. The previous code guarded against `nil` (Maybe it wasn't necessary or the test coverage we have isn't sufficient, I don't know)

Reply:
Ah...ok, but `@normalized_host` can be empty string, so it is proper way?

```ruby
if @normalized_host && !@normalized_host.empty?
  @normalized_host.force_encoding(Encoding::UTF_8)
end
```

Reply:
Yeah

Reply:
What if `@normalized_host` is an empty string but not `UTF-8` encoded..?

Reply:
I think that it will not be have matter because url without host is wrong url

:thinking: 

it is may be have sense to add condition on emptiness at this line: https://github.com/sporkmonger/addressable/blob/master/lib/addressable/uri.rb#L1246